### PR TITLE
Perses Design System

### PR DESCRIPTION
**Pull request by eunicorn**
This PR moves everything related to theme to @perses-dev/components and adds colors from [design system](https://www.figma.com/file/D9VznEcxMYrhyVwSb6fLuf/Perses-Design?node-id=1%3A68&t=9EHzZKUDz2JVfLRv-0) 

As a result, the app is updated with new colors :) 

**Light Mode**
Before:
image

After:
image

**Dark Mode**
Before:
image

After:
image

### CODE REVIEW

File: ui/app/tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "extends": "../tsconfig.base.json",
-  "include": ["src"],
+  "include": ["src", "../components/src/theme/palette/colors/types.ts", "../components/src/theme/palette/colors/grey.ts", "../components/src/theme/palette/secondary.ts", "../components/src/theme/palette/primary.ts"],
```
Comment (LukeTillman) [inline ui/app/tsconfig.json:3]: This doesn't seem right. The `app` package shouldn't be building these files from the other package. Maybe this got updated automatically/accidentally when you moved the files?

File: ui/dashboards/src/views/ViewDashboard/DashboardApp.tsx [hunk 2]
```diff
@@ -39,6 +39,7 @@ export const DashboardApp = (props: DashboardAppProps) => {
   const [originalDashboard, setOriginalDashboard] = useState<DashboardResource | undefined>(undefined);
   const [isUnsavedDashboardDialogOpen, setUnsavedDashboardDialogIsOpen] = useState(false);
 
+  console.log('dashboard', dashboard);
```
Comment (LukeTillman) [inline ui/dashboards/src/views/ViewDashboard/DashboardApp.tsx:42]: Probably can remove.

File: ui/dashboards/src/components/DashboardToolbar/DashboardToolbar.tsx [hunk 3]
```diff
@@ -49,30 +49,34 @@ export const DashboardToolbar = (props: DashboardToolbarProps) => {
...
+              padding: (theme) => theme.spacing(0, 2, 2, 2),
             }}
           >
             <ErrorBoundary FallbackComponent={ErrorAlert}>
-              <TemplateVariableList initialVariableIsSticky={initialVariableIsSticky} />
+              <TemplateVariableList
+                initialVariableIsSticky={initialVariableIsSticky}
+                sx={{
+                  backgroundColor: ({ palette }) =>
+                    palette.mode === 'dark' ? palette.background.default : palette.background.paper,
```
Comment (LukeTillman) [inline ui/dashboards/src/components/DashboardToolbar/DashboardToolbar.tsx:77]: This seems a little weird that we're using different theme tokens here for the background in light vs dark. Maybe something to follow-up with @foxbat07 on?

**Review by LukeTillman — APPROVED** [review]
Just a couple minor things, but this looks great!

File: ui/app/tsconfig.json (hunk 1 above)
Comment (eunicorn) [reply ui/app/tsconfig.json]: oh yep reverted this change!

**Comment by juliepagano** [comment]
I noticed some of the headings in the app look really squished up against the top looking at the code running locally. Wondering if some spacing/font changes adjusted things just enough to look off.

<table>
	<tr>
		<th>before</th>
		<th>after</th>
	</tr>
	<tr>
		<td>Screen Shot 2022-11-22 at 10 59 57 AM</td>
		<td>Screen Shot 2022-11-22 at 11 00 36 AM</td>
	</tr>
</table>

**Comment by eunicorn** [comment]
> I noticed some of the headings in the app look really squished up against the top looking at the code running locally. Wondering if some spacing/font changes adjusted things just enough to look off.
> 
> before	after
> Screen Shot 2022-11-22 at 10 59 57 AM	Screen Shot 2022-11-22 at 11 00 36 AM

fixed, thanks :)

**Review by juliepagano — APPROVED** [review]
This looks great! 🎉

File: ui/components/src/InfoTooltip/InfoTooltip.tsx [hunk 4]
```diff
@@ -78,13 +78,13 @@ const StyledTooltip = styled(({ className, ...props }: MuiTooltipProps) => (
   <MuiTooltip {...props} classes={{ popper: className }} />
 ))(({ theme }) => ({
   [`& .${tooltipClasses.tooltip}`]: {
-    backgroundColor: theme.palette.grey[100],
+    backgroundColor: theme.palette.background.tooltip,
```
Comment (sjcobb) [inline ui/components/src/InfoTooltip/InfoTooltip.tsx:81]: Nice! I might try this in LineChart Tooltip later

File: ui/components/src/theme/index.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import type {} from './types/ThemeExtension';
+import { PersesColor } from './palette';
+
+export * from './theme';
+
+//Use Typescript interface augmentation to extend the MUI type definition
```
Comment (sjcobb) [inline ui/components/src/theme/index.ts:19]: Awesome you got this working, I remember getting pretty frustrated with it awhile back

**Review by sjcobb — APPROVED** [review]
🎉

**Review by Nexucis — COMMENTED** [review]
Pretty impressive :).

Just some side notes:

* When I ran `npm install` on your branch, `package-lock.json` has been modified.
* I was looking at the [Perses design](https://www.figma.com/file/D9VznEcxMYrhyVwSb6fLuf/Perses-Design?node-id=1%3A390) and when you are editing, the blue background is still there. Which is not the case in your PR: 
image

Is it intended ?

**Comment by Nexucis** [comment]
> This PR mainly focuses on adding new colors and moving the theme to @perses-dev/components. Will open another PR to update the app to look like the mock ups

Yes that makes sense, but I said that because on the main branch you have a background color that doesn't exist anymore in your PR

image

So that's why I'm asking if it's intended :). 

> This should be fine since npm install will modify package-lock if it installs different minor versions. Can you show me the diff?

After pulling your branch again, I don't have anymore a diff. So probably I have messed up my environment before playing with your branch. Sorry for the noise.

**Comment by eunicorn** [comment]
> Yes that makes sense, but I said that because on the main branch you have a background color that doesn't exist anymore in your PR

ohhh good catch, thank you! fixed it :)

**Review by Nexucis — APPROVED** [review]
nice, thank you @eunicorn :). It looks really cool with the new themes :)

**Pull request merged**